### PR TITLE
fwupd-bsd-packages-and-ci.md: Describe building packages

### PR DESCRIPTION
**Pull request by Asiderr**
Signed-off-by: Norbert Kamiński <norbert.kaminski@3mdeb.com>

### CODE REVIEW

File: blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+
+# References
+
+FreeBSD port: https://github.com/3mdeb/freebsd-ports/pull/1
+OpenBSD port: https://github.com/3mdeb/ports/pull/1
+NetBSD port: https://github.com/3mdeb/pkgsrc-wip/pull/1
+DragonflyBSD port: https://github.com/3mdeb/DPorts/pull/1
+fwupd upstream PR: https://github.com/fwupd/fwupd/pull/2874
+
```
Comment (Asiderr) [inline blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md:325]: @pkonkol Add Link to the CI job

File: blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+ports based build uses dependencies from binary packages available from `pkg`
+and there is no need to build or manually cache dependencies. Next thing we
+needed to do was obtain the up-to-date ports sources with fwupd. As a temporary
+workaround, until fwupd will be available in the freebsd ports upstream we
+clone our fork and use it as a base to build the package. The Makefile is made
+for the tagged release version, so there is a slight complication. To get past
+it we needed to `sed` through the file setting github related parameters
+accordingly with the branch from which CI was started.
+
+
```
Comment (Asiderr) [inline blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md:316]: @pkonkol too many blank lines

File: blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+generates the biggest number of problems was libusb which differs
+from distro to distro. Also, I need to mention that OpenBSD has no built-in
+D-Bus so it needs to be provided as `x11/dbus` port
+(the port documentation is poor).
+
+# Continuous Integration for FreeBSD package
+
+Running CI/CD scripts on FreeBSD is not the most common DevOps tasks, that's
+why we were lucky to find out that Github Actions already had a way to run
+something in FreeBSD withoud setting up your own local runner. It was also
```
Comment (Asiderr) [inline blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md:285]: without

File: blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+already used by fwupd upstream for performing build on multiple distros,
+fuzzing and verifying ABI.
+
+However the way in which we are able to use this FreeBSD on a shared runner is
```
Comment (Asiderr) [inline blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md:289]: However,

File: blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+very interesting. We used
+[vmactions/freebsd-vm](https://github.com/vmactions/freebsd-vm) which itself is
+based on MacOS shared runner which is the OS chosen in CI code.  MacOS is used
+here to start virtualbox in which FreeBSD machine is ran.  The commands we
```
Comment (Asiderr) [inline blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md:293]: is running

File: blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+specify for the continous integration script are executed through SSH and then
```
Comment (Asiderr) [inline blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md:294]: continuous

File: blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+the resulting output files are rsync'ed back to MacOS runner.  Even though it's
+based on workarounds it perfoms it task ok.
```
Comment (Asiderr) [inline blog/content/post/2021-03-15-fwupd-bsd-packages-and-ci.md:296]: performs

**Review by artur-rs — APPROVED** [review]

**Pull request merged**